ui/src/views/WorkerManager/WMEditWorkerPool/index.jsx

```diff
@@ -0,0 +1,223 @@
+      this.setState({
+        workerPool,
+        invalidProviderConfig: true,
+      });
+    }
+  };
+
+  handleCreateWorkerPool = () => {};
+
+  render() {
+    const { isNewWorkerPool, classes } = this.props;
+    const { workerPool, invalidProviderConfig } = this.state;
+
+    return (
+      <Dashboard
+        title={isNewWorkerPool ? 'Create Worker Pool' : 'Edit Worker Pool'}>
+        <List>
+          <ListItem>
+            <TextField
+              label="Name"
+              name="name"
+              error={
+                Boolean(workerPool.name) &&
+                !isWorkerTypeNameValid(workerPool.name)
+              }
+              onChange={this.handleInputChange}
+              fullWidth
+              value={workerPool.name}
+              margin="normal"
+            />
+          </ListItem>
+
+          <ListItem>
+            <TextField
+              label="Description"
+              name="description"
+              onChange={this.handleInputChange}
+              fullWidth
+              value={workerPool.description}
+              margin="normal"
+            />
+          </ListItem>
+
+          <ListItem>
+            <TextField
+              label="Owner's Email"
+              name="owner"
+              error={
+                Boolean(workerPool.owner) && !workerPool.owner.includes('@')
+              }
+              onChange={this.handleInputChange}
+              fullWidth
+              value={workerPool.owner}
+              margin="normal"
+            />
+          </ListItem>
+
+          <ListItem>
+            <FormControlLabel
+              control={
+                <Switch
+                  checked={workerPool.wantsEmail}
+                  onChange={this.handleSwitchChange}
+                  value="wantsEmail"
+                />
+              }
+              label="Email the owner about errors"
+            />
+          </ListItem>
+
+          <ListItem>
+            <FormLabel component="provider">Provider:</FormLabel>
+          </ListItem>
+          <List>
+            <ListItem>
+              <TextField
+                id="select-provider-type"
+                select
+                label="Type:"
+                helperText="Which service do you want to run your tasks in?"
+                value={workerPool.providerType}
+                name="providerType"
+                onChange={this.handleInputChange}
+                margin="normal">
+                {Object.keys(providers).map(p => (
+                  <MenuItem key={p} value={p}>
+                    {p}
+                  </MenuItem>
+                ))}
+              </TextField>
+            </ListItem>
+
+            <ListItem>
+              <TextField
+                label="Name"
+                value={workerPool.providerId}
+                name="providerId"
+                onChange={this.handleInputChange}
+                margin="normal"
+              />
+            </ListItem>
+          </List>
```

Comment:
Instead of indenting the provider name and provider type

![Screen Shot 2019-06-03 at 10 08 31 AM](https://user-images.githubusercontent.com/3766511/58808220-9da0a000-85e7-11e9-97b2-b15fc7301043.png)


we can put everything in one line:

![Screen Shot 2019-06-03 at 9 59 35 AM](https://user-images.githubusercontent.com/3766511/58807590-582fa300-85e6-11e9-80f2-65ec04ac831e.png)

```jsx
<Grid container>
  <Grid item xs={12} sm={6}>
    <ListItem>
      <TextField
        fullWidth
        label="Name"
        value={workerPool.providerId}
        name="providerId"
        onChange={this.handleInputChange}
      />
    </ListItem>
  </Grid>
  <Grid item xs={12} sm={6}>
    <ListItem>
      <TextField
        className={classes.dropdown}
        id="select-provider-type"
        select
        label="Type"
        helperText="Which service do you want to run your tasks in?"
        value={workerPool.providerType}
        name="providerType"
        onChange={this.handleInputChange}>
        {Object.keys(providers).map(p => (
          <MenuItem key={p} value={p}>
            {p}
          </MenuItem>
        ))}
      </TextField>
    </ListItem>
  </Grid>
</Grid>
```

Other edits in the code block:
* I removed `margin="normal"` from both text fields in order to bring them closer to the list subheader "Provider".'
* `s/Type:/Type/`
* Added `fullWidth` to the first TextField

Reply:
> we can put everything in one line

how is this better than indenting? If one uses trackpad or a mouse, indented fields require less movement as they are grouped more closely together. And I just like the indentation better :)

Reply:
It might not matter: the user doesn't get to specify the provider type here, just the provider ID.  There's no such thing as "provider name".

Reply:
I don't think it is necessary to use indentation when there is a clear subheader that does the job of separation. Also, I think Indented text fields don't sit well on mobile.

![Screen Shot 2019-06-03 at 12 08 24 PM](https://user-images.githubusercontent.com/3766511/58816841-54a51780-85f8-11e9-99d8-334232029fe6.png)

Reply:
> the provider type here, just the provider ID. There's no such thing as "provider name".

We don't have to leak the names of DB fields into the UI, and the reason I didn't change it - in my mind ID is usually a numeric or a UUID-type of string. My intention was to make it clearer that user can enter an understandable thing in there, like a name.

That said, if you think "provider ID" is better than "provider name", I can replace that.

Reply:
> Also, I think Indented text fields don't sit well on mobile.

Indeed! I guess we don't use media queries in our code?

Incidentally, The whole thing doesn't look very good to me on mobile because of that padding or margin on the left.... _Should_ we use a media query for that? Looks like a waste of a lot of space to me

Reply:
We are pretty consistent on using "ID" throughout the UI, and on using IDs as unique identifiers.  Roles have "Role ID", clients have "Client ID", etc.

Reply:
> Incidentally, the whole thing doesn't look very good to me on mobile because of that padding or margin on the left

I think you can look at how the HookForm component does it, but I think it might involve removing the `margin` prop being set on the TextFields.

![Screen Shot 2019-06-03 at 1 08 18 PM](https://user-images.githubusercontent.com/3766511/58820561-e9ac0e80-8600-11e9-8a45-18f0494fd5c5.png)

Reply:
> but I think it might involve removing the margin prop being set on the TextFields.

No, the margin comes from the `List` element

Reply:
so I removed the padding from the `List` and `ListItem` components so that it looked like the Hooks form. This looks good to me; what do you think?

<img width="365" alt="screen" src="https://user-images.githubusercontent.com/18102552/58825036-61bf0800-85f2-11e9-8fcc-e5607fca92b8.png">

Reply:
Sorry, updated version:

<img width="361" alt="screen" src="https://user-images.githubusercontent.com/18102552/58825564-9b444300-85f3-11e9-905f-dede1814066d.png">

Reply:
Looks a lot better than before. I still think the indentation is more suitable for things like tree views, radio buttons, etc. I think Indentation is not necessary for inputs when there is a subheader above it. If we are trying to add more emphasis to it, then we can try adding more vertical spacing on top of list subheaders. Let me know if that does the job.

Reply:
> If we are trying to add more emphasis to it

Yes, that's exactly what I am trying to do. Subheaders don't work for me because they're small font (even smaller than those label/placeholder hybrids), and the whole thing just looks like a uniform wall of text to me.

I'll try experimenting with vertical spacing

Reply:
This spacing works for me:

<img width="413" alt="screen" src="https://user-images.githubusercontent.com/18102552/58833477-f7b05e00-8605-11e9-871a-87c11564e771.png">

Reply:
(it's in the last commit I pushed)